### DISCUSSION TITLE
runtime or build time dependencies

### DISCUSSION THREAD

**Discussion by silentnoodlemaster** [mhx/dwarfs#187] [Q&A] [answered]
is the flac library needed needed as a runtime dependency or is it only during build?

**Comment by mhx** [comment, marked as answer]
It's a run-time dependency. You can explicitly build with `-DENABLE_FLAC=OFF` if you want to drop FLAC support. The prebuilt binaries from the release page are all built statically linked with FLAC.

**Reply by mhx** [reply]
That being said, it'd be great if FLAC support was actually enabled, as being able to transparently compress PCM audio files as part of a DwarFS image is one of the major features of the new release. I just checked the [arch package](https://aur.archlinux.org/packages/dwarfs) and it looks like it doesn't have a dependency on FLAC.

**Reply by silentnoodlemaster** [reply]
yeah i don't know that maintainer, but have had problem using that package in the past. now i maintain a separate repository that also includes dwarfs, here is the PKGBUILD that i just updated for 0.8.0 https://github.com/jc141x/rumpowered-packages/blob/master/pkgbuilds/dwarfs/PKGBUILD

**Reply by mhx** [reply]
FWIW, I just built and installed the dwarfs AUR package and it works flawlessly.

**Reply by silentnoodlemaster** [reply]
tbh, I can't even remember what the reason was for making a new PKGBUILD, in their current state they are almost identical

**Comment by silentnoodlemaster** [comment]
at runtime, would it be libFLAC++.so , or the flac binary. and build time the header files from FLAC++ ?

**Reply by mhx** [reply]
The shared object at runtime. May I ask what you are trying to do?

**Reply by silentnoodlemaster** [reply]
I'm just packaging, for fedora and arch. so just trying to figure out which dependencies go where

**Reply by mhx** [reply]
Awesome, let me know if you need more information!

**Reply by silentnoodlemaster** [reply]
I'm having a small problem with python-mistletoe, fedora does not package it, so i gotta figure out a way around that

**Reply by mhx** [reply]
Pass `-DWITH_MAN_OPTION=OFF` to cmake. This disables built-in man pages, which is perfectly fine when installing with a package manager.
